app/controllers/frontend_controller.rb

```diff
@@ -168,4 +170,15 @@ def render_wrong_domain_error
   def set_display_currency
     ThreeScale::MoneyHelper.display_currency = current_account.currency if current_account
   end
+
+  def quickstarts_presenter
+    current_api = if @backend_api&.id.present?
+                    @backend_api
+                  elsif @service&.id.present?
+                    @service
+                  else
+                    current_account.default_service
+                  end
+    @quickstarts_presenter ||= QuickstartsPresenter.new(current_api)
+  end
```

Comment:
Stylistic comment, thoughts on short circuiting if `quickstarts_presenter` exists? 
 This then avoids the logic around `current_api` unnecessarily on multiple invocations of this.

Also not sure if you can drop the `present?` call and simply check for the existence of the `id`?

```ruby
def quickstarts_presenter
  return @quickstarts_presenter if @quickstarts_presenter

  current_api = if @backend_api&.id
                  @backend_api
                elsif @service&.id
                  @service
                else
                  current_account.default_service
                end
  @quickstarts_presenter = QuickstartsPresenter.new(current_api)
end
```

Reply:
I like to use `present` for more readability but I think it's redundant in this case.